File: Backend/Backend/Models/FrontendIntegrationModels/ThesisDetails.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Backend.Models.FrontendIntegrationModels
{
    public class ThesisDetails
    {
        public string thesisName;
        public string stateName;
        public DateTime lastActionDate;
        public float? ThesisGrade;
        public float? StudyGrade;
        public float? DefenseGrade;
        public string graduateFile;
        public string promoterFile;
        public List<ThesisComment> comments; 
    }
}

File: UI/src/app/models/backend-integration-models/ThesisDetails.ts
import { ThesisCommentDetails } from "./ThesisCommentDetails";

export interface ThesisDetails{
    thesisName: string;
    stateName: string;
    lastActionDate: Date;
    defenseGrade: string;
    studyGrade: string;
    thesisGrade: string;
    graduateFile: number;
    promoterFile: number;
    comments: Array<ThesisCommentDetails>
}